travel/src/test/java/com/tx/travel/service/DailyAllowanceServiceTest.java

```diff
@@ -0,0 +1,253 @@
+
+  @DisplayName(
+      "given daily allowance"
+          + "when daily allowance is updated"
+          + "then daily allowance is updated successfully")
+  @Test
+  public void updateDailyAllowance_success() {
+    final Long id = 1L;
+    final BigDecimal testAmount = new BigDecimal(100);
+    final String testRegion = "TestRegion1";
+
+    DailyAllowance dailyAllowance =
+        DailyAllowance.builder().id(id).region(testRegion).amount(testAmount).build();
+
+    when(dailyAllowanceRepository.findByRegion(testRegion))
+        .thenReturn(Optional.of(DailyAllowance.builder().region(testRegion).build()));
+    when(dailyAllowanceRepository.save(dailyAllowance)).thenReturn(dailyAllowance);
+
+    final DailyAllowance result = sut.updateDailyAllowance(dailyAllowance);
+    // assertNotNull(result);
+    verify(dailyAllowanceRepository, times(1)).save(dailyAllowanceArgumentCaptor.capture());
+    DailyAllowance result1 = dailyAllowanceArgumentCaptor.getValue();
+    assertThat(result.getAmount()).isEqualTo(dailyAllowance.getAmount());
+  }
+
+  @DisplayName(
+      "given daily allowance"
+          + "when daily allowance is searched by id"
+          + "then daily allowance is returned successfully")
+  @Test
+  public void findById_success() {
+    final Long id = 1L;
+    final BigDecimal testAmount = new BigDecimal(100);
+    final String testRegion = "TestRegionFindById";
+
+    DailyAllowance dailyAllowance =
+        DailyAllowance.builder().id(id).region(testRegion).amount(testAmount).build();
+
+    when(dailyAllowanceRepository.findById(id))
+        .thenReturn(Optional.of(DailyAllowance.builder().id(id).build()));
+
+    final DailyAllowance result = sut.findById(id);
+    assertNotNull(result);
+  }
+
+  @DisplayName(
+      "given daily allowance"
+          + "when daily allowance is searched by id"
+          + "then throw DailyAllowanceNotFoundException")
+  @Test
+  public void findById_DailyAllowanceNotFoundException() {
+    final Long id = 1L;
+    final BigDecimal testAmount = new BigDecimal(100);
+    final String testRegion = "TestRegionFindById";
+    DailyAllowanceNotFoundException e = new DailyAllowanceNotFoundException(testRegion);
+
+    when(dailyAllowanceRepository.findById(id)).thenThrow(e);
+
+    assertThrows(DailyAllowanceNotFoundException.class, () -> sut.findById(id));
+  }
+
+  @DisplayName(
+      "given daily allowance"
+          + "when daily allowance is added"
+          + "then daily allowance is added successfully")
+  @Test
+  public void addDailyAllowance_success() {
+    final Long id = 1L;
+    final BigDecimal testAmount = new BigDecimal(100);
+    final String testRegion = "TestRegionAdd";
+
+    DailyAllowance dailyAllowance =
+        DailyAllowance.builder().id(id).region(testRegion).amount(testAmount).build();
+
+    when(dailyAllowanceRepository.save(dailyAllowance)).thenReturn(dailyAllowance);
+
+    assertDoesNotThrow(() -> sut.addDailyAllowance(dailyAllowance));
```

Comment:
maybe you can assert if returned daily allowance from save method matches the one you provided